app/src/main/java/ch/epfl/skysync/screens/crewpilot/Flight.kt

```diff
@@ -181,20 +183,21 @@ fun FlightScreen(
     // Cleanup function to stop receiving location updates when the composable is disposed.
     onDispose { fusedLocationClient.removeLocationUpdates(locationCallback) }
   }
+  println(
+      "Loading: $loading isPilot: ${inFlightViewModel.isPilot()} currentFlight: ${currentFlight?.id} startableFlights: ${startableFlights.size}")
+  println("Permission: ${locationPermission.status.isGranted}")
   if (!locationPermission.status.isGranted) {
     Box(modifier = Modifier.fillMaxSize().padding(16.dp), contentAlignment = Alignment.Center) {
       Column(horizontalAlignment = Alignment.CenterHorizontally) {
         Text("Access to location is required to use this feature.")
         Text("Please enable location permissions in settings.")
       }
     }
-  } else if (personalFlights == null) {
+  } else if (loading) {
     LoadingComponent(isLoading = true, onRefresh = {}) {}
-  } else if (personalFlights!!.isEmpty()) {
-    ShowFlightToStart(navController = navController, flight = null) {}
-  } else if (currentFlightId == null) {
-    ShowFlightToStart(navController, personalFlights!!.first()) {
-      inFlightViewModel.setFlightId(it)
+  } else if (currentFlight == null) {
+    ShowFlightToStart(navController, startableFlights.firstOrNull()) {

```

Comment:
if we have two flights (AM and PM) it might choose one of the two at random?

Reply:
Doesn't happen anymore as we only consider flight in current time slot (either AM or PM)